### PR TITLE
feat(argo-cd): Bump dex dependency for CVE-2021-3450

### PR DESCRIPTION
**Pull request by duncaan**
Updates the dex image to address CVE-2021-3450.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [ ] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Review by mbevc1 — APPROVED** [review]

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -21,4 +21,4 @@ dependencies:
     condition: redis-ha.enabled
 annotations:
   artifacthub.io/changes: |
-    - "[Fixed]: redis-ha.nameOverride / redis-ha.fullnameOverride breaks the ArgoCD helm chart"
+    - "[Updated]: Updated dex image 2.28.1 -> 2.30.0"
```
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:24]: `[Updated]` is not in our documentation. Please use one of whats documented :-) I think `[Security]` fits best

> Changelog
> We want to deliver transparent chart releases for our chart consumers. Therefore we require a changelog per new chart release.
> 
> Changes on a chart must be documented in a chart specific changelog in the Chart.yaml Annotation Section. For every new release the entire artifacthub.io/changes needs to be rewritten. Each change requires a new bullet point following the pattern - "[{type}]: {description}". You can use the following template:
> 
> name: argo-cd
> version: 3.4.1
> ...
> annotations:
>   artifacthub.io/changes: |
>     - "[Added]: Something New was added"
>     - "[Changed]: Changed Something within this chart"
>     - "[Changed]: Changed Something else within this chart"
>     - "[Deprecated]: Something deprecated"
>     - "[Removed]: Something was removed"
>     - "[Fixed]: Something was fixed"
>     - "[Security]": Some Security Patch was included"

File: charts/argo-cd/values.yaml [hunk 2]
```diff
@@ -233,7 +233,7 @@ dex:
 
   image:
     repository: ghcr.io/dexidp/dex
-    tag: v2.28.1
+    tag: v2.30.0
```
Comment (mkilchhofer) [inline charts/argo-cd/values.yaml:236]: I am unsure if we should update dex to a newer verseion than what's bundled within the official kustomize manifests over there: https://github.com/argoproj/argo-cd/blob/v2.0.5/manifests/base/dex/argocd-dex-server-deployment.yaml

Everyone is free to use a newer one and can override it through values. But the app release is tested with the versions used in the kustomize manifests.
Other opinions @mbevc1 @oliverbaehler @davidkarlsen @seanson ?

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

File: charts/argo-cd/values.yaml (hunk 2 above)
Comment (mbevc1) [reply charts/argo-cd/values.yaml:236]: Hm, yeah that seems sensible approach until upstream doesn't support this combo.

**Comment by mbevc1** [comment]
Are there now more changes than intended in this PR?

**Pull request closed without merging**